### PR TITLE
Fixes #18096: Technique from AST

### PR DESCRIPTION
**Pull request by gpoblon**
https://issues.rudder.io/issues/18096

### CODE REVIEW

**Comment by gpoblon** [comment]
Lot of json test files updated, just ignore this part.
svg part is just reverting changes form previous PR (Compiler to Generator)

File: rudder-lang/src/parser/token.rs [hunk 1]
```diff
@@ -55,6 +55,34 @@ impl<'src> Token<'src> {
         &self.val.extra
     }
 
+    pub fn from_lib(&self) -> bool {
+        // Should do the trick for now, since we have no context in here to get libs/std/ filenames dynamically
```
Comment (amousset) [inline rudder-lang/src/parser/token.rs:59]: we could maybe add a tag to the states coming from the lib in the AST

File: rudder-lang/src/technique/from_ast.rs [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+            })
+            .collect::<Vec<&ResourceDef>>();
+        let resource = resources
+            .get(0)
+            .expect("There should only be one resource definition matching the technique");
+
+
+        let meta = &resource.metadata;
+        
+        let interpolated_parameters: Vec<InterpolatedParameter> = meta.get("parameters")
```
Comment (amousset) [inline rudder-lang/src/technique/from_ast.rs:38]: Could you run rustfmt? Ideally it should be run by your text editor on save.

**Comment by gpoblon** [comment]
OK, squash merging this PR

**Pull request merged**